### PR TITLE
Option to wrap union members in ManuallyDrop

### PR DESCRIPTION
**Pull request by poliorcetics**
Adds the option to wrap members of union where at least one is Non-Copy
in `::core::mem::ManuallyDrop` instead of the generated `__BindgenUnionWrapper`,
with the approriate CLI and API options (default, regex for `ManuallyDrop`/
`__BindgenUnionWrapper`).

### CODE REVIEW

**Comment by poliorcetics** [comment]
Failing test is due to a connection error.

File: src/codegen/mod.rs [hunk 1]
```diff
@@ -3357,6 +3404,52 @@ impl std::str::FromStr for AliasVariation {
     }
 }
 
+/// Enum for how non-Rust unions should be translated.
```
Comment (emilio) [inline src/codegen/mod.rs:3407]: Calling this non-rust unions, yet generating rust unions feels a bit weird... Maybe this could be `NonCopyUnionStyle` or so?

**Review by emilio — COMMENTED** [review]
Thanks, this looks generally fine (sorry for the lag here). But naming seems a bit unfortunate, or at least while reviewing it I've been weirded out about it a bit.

What do you think about the suggestion below? Or maybe there are other better options?

**Comment by poliorcetics** [comment]
Sorry for the long wait, I rebased and fixed your comment

File: src/codegen/mod.rs [hunk 2]
```diff
@@ -1511,7 +1550,7 @@ impl<'a> FieldCodegen<'a> for BitfieldUnit {
         struct_layout: &mut StructLayoutTracker,
         fields: &mut F,
         methods: &mut M,
-        _: (),
+        parent_canonical_name: Self::Extra,
```
Comment (emilio) [inline src/codegen/mod.rs:1553]: Why is this needed? Bitfields should always be copy, right?

**Review by emilio — APPROVED** [review]
Sorry for the massive lag. This looks good but we should simplify the bitifeld handling because by definition bitfields need to be copy. I can do that in a follow-up.
